celery-stubs/worker/request.pyi

```diff
@@ -127,6 +127,7 @@ class Request:
     def revoked(self) -> bool: ...
     def send_event(self, type: str, **fields: Any) -> None: ...
     def on_accepted(self, pid: str, time_accepted: float) -> None: ...
+    def on_timeout(self, soft: bool, timeout: int) -> None: ...
```

Comment:
Timeouts values are inconsistently typed in celery-types, but I suspect most paths accept both int and float values?

```suggestion
    def on_timeout(self, soft: bool, timeout: int | float) -> None: ...
```